### PR TITLE
Upload files after application deadline

### PR DESCRIPTION
**Pull request by kdmnk**
And sends email to the committee members (also if the notes are changed)

### CODE REVIEW

**Review by horcsinbalint — CHANGES_REQUESTED** [review]
I think these changes might trigger rate limiting for e-mail sending. We probably have a limit of 500 mails / 24 hours.
A better solution would be to send the same email with multiple recipients (the limit is probably 100 recipients / email via SMTP).

**Comment by BertalanD** [comment]
I'm a bit conflicted about sending emails for notes, but if you see a use case for it, sure. But please clarify in the helper text that an email will be sent.

**Comment by kdmnk** [comment]
> I'm a bit conflicted about sending emails for notes, but if you see a use case for it, sure. But please clarify in the helper text that an email will be sent.

True, I wanted to clarify it.
As far as I saw, the notes are not used that much, maybe to state some changes to the application and alike. For those, an email is nice but I can remove the feature.  You know better.

**Comment by BertalanD** [comment]
> As far as I saw, the notes are not used that much, maybe to state some changes to the application and alike. For those, an email is nice but I can remove the feature. You know better.

Keeping it as a way to urgently notify application committee members about important changes to an application sounds good to me.

**Comment by kdmnk** [comment]
I've used bcc now to send emails but I have issues testing it and I don't have more time now. Please test this on staging and fix if needed.

File: app/Http/Controllers/Auth/AdmissionController.php [hunk 1]
```diff
@@ -142,25 +145,30 @@ public function show(Application $application): View
...
+            $oldValue = $application->note;
             $application->update(['note' => $request->input('note')]);
-        } elseif ($newStatus) {
+            Mail::bcc($application->committeeMembers())->queue(new ApplicationNoteChanged(user(), $application, $oldValue));
+        }
+        if ($request->has('file')) {
             $this->authorize('editStatus', $application);
-            $application->update(['status' => $newStatus]);
+            $this->storeFile($request, $application->user);
+            Mail::bcc($application->committeeMembers())->queue(new ApplicationFileUploaded($request->get('name'), $application));
```
Comment (BertalanD) [inline app/Http/Controllers/Auth/AdmissionController.php:171]: This throws an exception if `committeeMembers` is empty. Please address it post-commit.

File: app/Http/Controllers/Auth/AdmissionController.php [hunk 2]
```diff
@@ -142,25 +145,30 @@ public function show(Application $application): View
...
-
-        $newStatus = $request->input('status_'.$application->user->id);
         if ($request->has('note')) {
+            $request->validate([
+                'note' => 'string',
+            ]);
+            $oldValue = $application->note;
             $application->update(['note' => $request->input('note')]);
-        } elseif ($newStatus) {
+            Mail::bcc($application->committeeMembers())->queue(new ApplicationNoteChanged(user(), $application, $oldValue));
```
[the remark this answers is not in the document]
Comment (BertalanD) [reply app/Http/Controllers/Auth/AdmissionController.php:166]: seems like a false positive? `user()` is used above too, without problems

**Review by BertalanD — APPROVED** [review]
I'm ready to merge and deploy this.

@viktorcsimma do you have anything to add before I do that?

**Comment by viktorcsimma** [comment]
I haven't checked it out thoroughly, but LGTM; thanks:)

**Comment by BertalanD** [comment]
Needed this fix post-commit: https://github.com/EotvosCollegium/mars/commit/7d93901ca25222553bfd4c0758279f66664aa4f3

**Pull request merged**